lib/compress/zstd_compress_internal.h

```diff
@@ -1077,6 +1081,88 @@ MEM_STATIC ZSTD_dictMode_e ZSTD_matchState_dictMode(const ZSTD_MatchState_t *ms)
             ZSTD_noDict;
 }
 
+/**
+ * Fractional window sizes can always be picked by the user explicitly
+ * setting ZSTD_c_windowFrac. This macro controls whether, when Zstd is
+ * picking a window size itself, it is allowed to pick a non-power-of-two
+ * window size.
+ *
+ * For now, this defaults to false.
+ */
+#ifndef ZSTD_WINDOW_ALLOW_PICKING_FRACTIONAL_SIZES
+#define ZSTD_WINDOW_ALLOW_PICKING_FRACTIONAL_SIZES 0
+#endif
+
+/**
+ * Return the window size described by the windowLog and windowFrac in the
+ * provided CParams.
+ */
+MEM_STATIC U32 ZSTD_windowSize(const ZSTD_CCtx_params* params) {
+    return (U32)(((8ull + params->windowFrac) << params->cParams.windowLog) >> 3);
+}
+MEM_STATIC U32 ZSTD_windowSizeLDM(const ldmParams_t* ldmParams) {
+    return (U32)(((8ull + ldmParams->windowFrac) << ldmParams->windowLog) >> 3);
+}
+
+/**
+ * Checks that the selected windowSize satisfies the inequality
+ * `srcSize <= windowSize <= srcSize * margin`, in the range where the window
+ * isn't pressed up against one of the hard bounds, where `margin` is either
+ * 1.125 or 2, depending on whether we're allowed to pick fractional window
+ * sizes.
+ */
+MEM_STATIC int ZSTD_windowLogAndFracAreMinimal(const ZSTD_CCtx_params* params, const U32 srcSize) {
+    const U32 lowerBound = MIN(srcSize, 1u << ZSTD_WINDOWLOG_MAX);
+#if ZSTD_WINDOW_ALLOW_PICKING_FRACTIONAL_SIZES
+    const U32 upperBound = MAX(lowerBound + (lowerBound >> 3), 1u << ZSTD_WINDOWLOG_ABSOLUTEMIN);
+#else
+    const U32 upperBound = MAX(2 * lowerBound - 1, 1u << ZSTD_WINDOWLOG_ABSOLUTEMIN);
+#endif
+    const U32 windowSize = ZSTD_windowSize(params);
+    if (windowSize < lowerBound) {
+        return 0;
+    }
+    if (windowSize > upperBound) {
+        return 0;
+    }
+    return 1;
+}
+
+/**
+ * Calculates the minimum legal window log and fraction that contain the
+ * provided source size.
+ */
+MEM_STATIC void ZSTD_setMinimalWindowLogAndFrac(ZSTD_CCtx_params* params, const U32 srcSize, const U32 minWindowLog) {
```

Comment:
Same question here,
`ZSTD_setMinimalWindowLogAndFrac()` seems only useful within `zstd_compress.c`.
What benefit is there to share it more widely within `zstd_compress_internal.h` ?

Note that `ZSTD_windowLogAndFracAreMinimal()` above seems to be a dependency of this function.